wslexe_windows.go

```diff
@@ -29,6 +29,58 @@ func terminate(distroName string) error {
 	return exec.Command("wsl.exe", "--terminate", distroName).Run()
 }
 
+// setAsDefault sets a particular distribution as the default one.
+//
+// It is analogous to
+//  `wsl.exe --set-default <distroName>`
+func setAsDefault(distroName string) error {
+	out, err := exec.Command("wsl.exe", "--set-default", distroName).CombinedOutput()
```

Comment:
TODO: Use registry rather than starting a whole process to set this field.

Reply:
Cannot do: editing the registry requires special permissions. I implemented this function via the registry and got:
```
failed to set "wsltesting_TestDistroSetAsDefault_93405798" as default distro: Access is denied.
```

Reply:
Maybe add it as a comment in the function for documentation and not wondering again why we can’t.

There are some policy-kit model I think to get special permissions, as `wsl.exe --set-default` can be run without being admin? (not really suitable for a package though as it means every process would need to register with those special perms)

Reply:
Oh no, `wsl.exe --set-default` works fine. What doesn't work is the TODO.

AFAIK all  `wsl.exe --set-default` does is change the field Lxss\DefaultDistribution in the Windows registry. So I tried to avoid spawning the wsl.exe process by modifying the registry directly. This is not allowed. So I fell back to the solution you see here.

Reply:
Yeah I think that is possible, the issue is the one you present. I think sticking to this sub-optimal solution is fine.

Reply:
agreed :)